cli/tools/test.rs

```diff
@@ -965,6 +1114,179 @@ impl TestReporter for PrettyTestReporter {
+    result: &TestResult,
+    elapsed: u64,
+  ) {
+    if let Some(mut case) = self.cases.get_mut(&description.id) {
+      case.status = Self::convert_status(result);
+      case.set_time(Duration::from_millis(elapsed));
+    }
+  }
+
+  fn report_uncaught_error(&mut self, _origin: &str, _error: Box<JsError>) {}
+
+  fn report_step_register(&mut self, _description: &TestStepDescription) {}
+
+  fn report_step_wait(&mut self, _description: &TestStepDescription) {}
+
+  fn report_step_result(
+    &mut self,
+    description: &TestStepDescription,
+    result: &TestStepResult,
+    _elapsed: u64,
+    _tests: &IndexMap<usize, TestDescription>,
+    _test_steps: &IndexMap<usize, TestStepDescription>,
+  ) {
+    let status = match result {
+      TestStepResult::Ok => "passed",
+      TestStepResult::Ignored => "skipped",
+      TestStepResult::Failed(_) => "failure",
+    };
+    // TODO: This currently doesn't work for nested steps, not sure how I'd represent them...
+    if let Some(case) = self.cases.get_mut(&description.parent_id) {
+      case.add_property(quick_junit::Property::new(
+        format!("step[{}]", status),
+        description.name.clone(),
+      ));
+    }
+  }
+
+  fn report_summary(
+    &mut self,
+    _elapsed: &Duration,
+    _tests: &IndexMap<usize, TestDescription>,
+    _test_steps: &IndexMap<usize, TestStepDescription>,
+  ) {
+  }
+
+  fn report_sigint(
+    &mut self,
+    tests_pending: &HashSet<usize>,
+    tests: &IndexMap<usize, TestDescription>,
+    _test_steps: &IndexMap<usize, TestStepDescription>,
+  ) {
+    for id in tests_pending {
+      if let Some(description) = tests.get(id) {
+        self.report_result(description, &TestResult::Cancelled, 0)
+      }
+    }
+  }
+
+  fn flush_report(
+    &mut self,
+    elapsed: &Duration,
+    tests: &IndexMap<usize, TestDescription>,
+    _test_steps: &IndexMap<usize, TestStepDescription>,
+  ) -> Result<(), TestReporterError> {
+    let mut suites: IndexMap<String, quick_junit::TestSuite> = IndexMap::new();
+    for (id, case) in &self.cases {
+      if let Some(test) = tests.get(id) {
+        suites
+          .entry(test.location.file_name.clone())
+          .and_modify(|s| {
+            s.add_test_case(case.clone());
+          })
+          .or_insert_with(|| {
+            quick_junit::TestSuite::new(test.location.file_name.clone())
+              .add_test_case(case.clone())
+              .to_owned()
+          });
+      }
+    }
+
+    let mut report = quick_junit::Report::new("deno test");
+    report.set_time(*elapsed).add_test_suites(
+      suites
+        .values()
+        .cloned()
+        .collect::<Vec<quick_junit::TestSuite>>(),
+    );
+    let report_str = report.to_string()?;
+    let report_bytes = report_str.as_bytes();
+
+    if self.path == "-" {
+      std::io::stdout().write_all(report_bytes)?;
+    } else {
+      let mut file = std::fs::File::create(self.path.clone())?;
+      file.write_all(report_bytes)?;
```

Comment:
Please use `.with_context()` here that prints out something like "Failed to write Junit test report to '<path>'"

Reply:
Resolved in [75a9e7a](https://github.com/denoland/deno/pull/19747/commits/75a9e7a64ab371501294aa0e61fe0a682d6b9dbd)